src/App.tsx

```diff
@@ -9,8 +11,10 @@ import type { ActivePage } from '@/types/state.ts';
 import { getCartTotalAmount } from '@/utils/getCartTotalAmount.ts';
 
 function App() {
+    const isPreference = window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Consider moving this function into a separate file where you will find out what is user device theme and return the correct theme 

```
export const getUserTheme = () => {
    // Get preferred theme from local storage
    const userTheme = localStorage.getItem('theme');

    if (userTheme ) {
        return userTheme ;
    }

    const isDeviceUseDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches;

    const theme =  isDeviceUseDarkMode ? 'dark' : 'light';

    // Save preferred theme to local storage
    localStorage.setItem('theme', theme);

    return theme;
};
```